superset/constants.py

```diff
@@ -189,3 +189,32 @@ class CacheRegion(str, Enum):
     DEFAULT = "default"
     DATA = "data"
     THUMBNAIL = "thumbnail"
+
+
+legacy_charts = [
+    "line",
+    "dist_bar",
+    "area",
+    "time_table",
+    "histogram",
+    "time_pivot",
+    "heatmap",
+    "dual_line",
+    "line_multi",
+    "treemap",
+    "sunburst",
+    "sankey",
+    "mapbox",
+    "rose",
+    "bubble",
+    "horizon",
+    "compare",
+    "partition",
+    "event_flow",
+    "world_map",
+    "paired_ttest",
+    "para",
+    "bullet",
+    "chord",
+    "pivot_table",
+]
```

Comment:
I think a more dynamic approach would be to check if there's a query context for a given slice. If there is, assume it's the new type, and if there isn't, assume it's legacy.

Reply:
@villebro Right. But there are scenarios where query context are not found even for new charts. Like When you Import a chart query context is not found. it is handled in Alert and reports using a function called "update_query_context". I found no other way to differentiate between charts. Please suggest if you have any other idea.